pkg/apis/serving/k8s_validation.go

```diff
@@ -189,12 +189,12 @@ func validateEnvValueFrom(ctx context.Context, source *corev1.EnvVarSource) *api
 	return apis.CheckDisallowedFields(*source, *EnvVarSourceMask(source, features.FieldRef != config.Disabled))
 }
 
-func validateEnvVar(ctx context.Context, env corev1.EnvVar) *apis.FieldError {
+func validateEnvVar(ctx context.Context, env corev1.EnvVar, isSidecar bool) *apis.FieldError {
 	errs := apis.CheckDisallowedFields(env, *EnvVarMask(&env))
 
 	if env.Name == "" {
 		errs = errs.Also(apis.ErrMissingField("name"))
-	} else if reservedEnvVars.Has(env.Name) {
+	} else if (reservedEnvVars.Has(env.Name) && !isSidecar) || (reservedEnvVars.Has(env.Name) && env.Name != "PORT") {
```

Comment:
this probably requires a comment.

Reply:
Perhaps we should split `reservedEnvVars` into two sets:
1. Those that are set on all containers,
2. Those that are set on the main container.

Reply:
I thought of that but there is overlap and the diff is just one var.